Initial TS-7553-V2 support

ts-kris opened the pull request:
Support for TS-7553-V2 in 5.10

This adds its own defconfig with the intent that it will likely be merged in to a more singular defconfig in the future. However, to ensure a point in time where everything is tested and support, that is included in this PR as well.

This also adds support for the LCD as a framebuffer as it was in older kernels. Long term, the goal is to be able to support the monochrome LCD as a DRM supported panel with backlight, etc. The goal for this PR however is basic support of all features.

Additionally, it has been observed that microSD access is a bit slow here. Some optimizations and improvements can come later to speed that up.

ts-kris committed:
video: fbdev: st7565p: Add kernel driver half for SPI LCD

[ commit 2121d7fae0f621109d24febd52841abe324fe874 in 4.9.y ]

Bring in support for st7565p LCD using the driver shim and userspace
companion tool.

Signed-off-by: Kris Bahnsen <kris@embeddedTS.com>

ts-kris committed:
ARM: dts: imx6ul: Add TS-7553-V2 support

Add initial support of the i.MX6UL based TS-7553-V2 platform.

Signed-off-by: Kris Bahnsen <kris@embeddedTS.com>

randytsi369 reviewed (dismissed):
No issues found during review. Not tested.

ts-kris committed:
ARM: ts_defconfig: Changes for TS-7553-V2 support

Specifically adds IIO drivers, mono LCD driver, adds SNVS power
key as a builtin for compatibility with `keypad-test` demo in
ts7553v2-utils.git.

The framebuffer console is needed for TPC platforms, it causes
some issues in the TS-7553-V2 that will be resolved later.

Also removes drivers that would never be used, moves a few from
builtin to module, adds some other features/drivers in as a
module for potential future support.

Signed-off-by: Kris Bahnsen <kris@embeddedTS.com>

ts-kris commented:
Touches ts_defconfig to do the following:
- Move many things from builtins to modules, e.g. touchscreen drivers (note that this needs commit 25c6e79802a6d5e929d0691ee138f393513f8721 to work on resistive TPCs), audio codecs, IIO drivers, etc.
- Remove useless drivers, e.g. any non PCIe or USB network device driver that is not used on any of our platforms, filesystem drivers for ancient/unrelated devices, NAND specific drivers, regulator drivers, some media drivers, etc.
- Add in many new features from tsimx6ul_defconfig in 4.9 back here as modules, e.g. ipv6 netfilter
- Remove framebuffer console, necessary for TS-7553-V2 use of mono LCD but this may break uses of TPC where console should be present on the FB. Also add the SNVS powerkey driver as a built in to maintain the TS-7553-V2 demo keypad input. This can be removed/modularized later once the mono LCD gets a DRM panel driver.
- Add in driver support for TS-7553-V2 IIO

I did testing of the defconfig changes along the way under TS-4900, TS-7970, and TS-7990. Everything works as expected but I have not re-run through all of the testing again after doing this final squash. The last testing I did was with framebuffer console enabled and SNVS power key as a module. I don't suspect any issues (aside from resistive TPCs as I noted above) but will re-test things tomorrow.
